api/v1beta1/verticadb_types.go

```diff
@@ -217,6 +217,15 @@ type VerticaDBSpec struct {
 	// This value cannot change after the initial creation of the VerticaDB.
 	KSafety KSafetyType `json:"kSafety,omitempty"`
 
+	// +kubebuilder:default:=false
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// When set to true, the webhook will calculate the k-safety value
+	// based on the number of primary nodes in the cluster;
+	// otherwise, the calculation will be based on the number of all nodes
+	// in the cluster.
+	RelaxKSafetyCheck bool `json:"relaxKSafetyCheck,omitempty"`
```

Comment:
you don't need to add a new field to the CR spec, only the annotation, see how we do with vcluster-ops annotation.